src/util.rs

```diff
@@ -0,0 +1,10 @@
+use clap::App;
+pub trait ApplyToApp<'a, 'b> {
+    fn apply(self, f: fn(App<'a, 'b>) -> App<'a, 'b>) -> App<'a, 'b>;
+}
+
+impl<'a, 'b> ApplyToApp<'a, 'b> for App<'a, 'b> {
+    fn apply(self, f: fn(App<'a, 'b>) -> App<'a, 'b>) -> App<'a, 'b> {
```

Comment:
the crab language is hard sometimes

Reply:
i love the smell of lifetimes in the morning...

Reply:
This apply thing would be pretty useful for cleaning up main. I added it since I wanted to re-use the setup for the command line arguments in my test

Reply:
yeah i left a comment, def think this should become a trait in a future pr